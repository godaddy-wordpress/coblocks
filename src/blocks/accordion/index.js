/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
