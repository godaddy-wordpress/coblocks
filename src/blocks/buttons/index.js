/**
 * THIS BLOCK IS DEPRECRATED. IT WILL NOT SHOW UP IN THE INSERTER ANYMORE.
 * SEE /src/js/deprecations/deprecate-coblocks-buttons.js
 */

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { button as icon } from '@wordpress/icons';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Buttons', 'coblocks' ),
	/* translators: block description */
	description: __( 'Prompt visitors to take action with multiple buttons, side by side.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'link', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
