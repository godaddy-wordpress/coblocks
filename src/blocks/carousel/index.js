/**
 * WordPress dependencies
 */
import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import BlockEdit from './block-edit';
import metadata from './block.json';

const settings = {
	edit: BlockEdit,
	save: () => (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	),
};

const name = metadata.name;
export { name, metadata, settings };
