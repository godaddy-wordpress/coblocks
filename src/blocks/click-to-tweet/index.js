/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import { Icon } from '@wordpress/components';
import { TwitterIcon as icon } from '@godaddy-wordpress/coblocks-icons';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	deprecated,
	/* translators: block description */
	description: __( 'Add a quote for readers to tweet via Twitter.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			content: __( 'The easiest way to promote and advertise your blog, website, and business on Twitter.', 'coblocks' ),
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'share', 'coblocks' ),
		/* translators: block keyword */
		__( 'twitter', 'coblocks' ),
	],
	save,
	/* translators: block name */
	title: __( 'Click to Tweet', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
