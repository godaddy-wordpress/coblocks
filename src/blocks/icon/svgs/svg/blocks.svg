<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>