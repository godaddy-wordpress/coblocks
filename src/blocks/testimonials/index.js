/**
 * External dependencies
 */
import { TestimonialIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * WordPress dependencies.
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';

/**
 * Internal dependencies.
 */
import edit from './edit';
import metadata from './block.json';
import save from './save';
import { blockStyles as styles } from './utils';

/**
 * Block constants.
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Display a list of testimonials.', 'coblocks' ),
	edit,
	example: {
		innerBlocks: [
			{
				attributes: {
					description: 'Lorem ipsum dolor sit amet, consec tetur adipiscing elit. Nam condimentum tempus diam, ultricies sollicitudin erat facilisis eget. Vestibulum rhoncus dui vel eros laoreet consectetur.',
					name: 'John Doe',
					title: 'Senior Director',
				},
				name: 'coblocks/testimonial',
			},
			{
				attributes: {
					description: 'Lorem ipsum dolor sit amet, consec tetur adipiscing elit. Nam condimentum tempus diam, ultricies sollicitudin erat facilisis eget. Vestibulum rhoncus dui vel eros laoreet consectetur.',
					name: 'John Doe',
					title: 'Senior Director',
				},
				name: 'coblocks/testimonial',
			},
		],
	},
	getEditWrapperProps( atts ) {
		return { 'data-columns': atts.columns };
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'reviews', 'coblocks' ),
		/* translators: block keyword */
		__( 'testimonials', 'coblocks' ),
	],
	save,
	styles,
	/* translators: block name */
	title: __( 'Testimonials', 'coblocks' ),
};

export { name, category, metadata, settings };
