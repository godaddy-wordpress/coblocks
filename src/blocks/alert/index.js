/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
