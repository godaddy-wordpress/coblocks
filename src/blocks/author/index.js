/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
