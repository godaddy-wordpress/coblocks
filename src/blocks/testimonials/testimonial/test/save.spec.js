/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( 'coblocks/testimonial', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render empty when Name is missing', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with content', () => {
		block.attributes.name = 'Some name';
		block.attributes.role = 'Some role';
		block.attributes.text = 'Some text';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Some name' );
		expect( serializedBlock ).toContain( 'Some role' );
		expect( serializedBlock ).toContain( 'Some text' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with the "conversation" style', () => {
		block.attributes.name = 'Some name';
		block.attributes.styleName = 'conversation';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Some name' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with the "horizontal" style', () => {
		block.attributes.name = 'Some name';
		block.attributes.styleName = 'horizontal';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Some name' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with image', () => {
		block.attributes.name = 'Some name';
		block.attributes.url = 'http://some.url/someimage.jpg';
		block.attributes.showImage = true;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'http://some.url/someimage.jpg' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with focal point on image', () => {
		block.attributes.name = 'Some name';
		block.attributes.url = 'http://some.url/someimage.jpg';
		block.attributes.focalPoint = { x: 0.5, y: 1 };
		block.attributes.showImage = true;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'object-position:50% 100%' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
