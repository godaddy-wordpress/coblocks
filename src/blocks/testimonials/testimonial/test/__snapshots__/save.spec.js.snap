// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/testimonial should render empty when Name is missing 1`] = `"<!-- wp:coblocks/testimonial /-->"`;

exports[`coblocks/testimonial should render with content 1`] = `
"<!-- wp:coblocks/testimonial -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4><p class=\\"wp-block-coblocks-testimonial__heading__role\\">Some role</p></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\">Some text</p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with focal point on image 1`] = `
"<!-- wp:coblocks/testimonial {\\"focalPoint\\":{\\"x\\":0.5,\\"y\\":1},\\"showImage\\":true} -->
<div class=\\"wp-block-coblocks-testimonial\\"><figure class=\\"wp-block-coblocks-testimonial__image\\" style=\\"height:100px;width:100px\\"><img alt=\\"\\" src=\\"http://some.url/someimage.jpg\\" style=\\"object-position:50% 100%\\"/></figure><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with image 1`] = `
"<!-- wp:coblocks/testimonial {\\"showImage\\":true} -->
<div class=\\"wp-block-coblocks-testimonial\\"><figure class=\\"wp-block-coblocks-testimonial__image\\" style=\\"height:100px;width:100px\\"><img alt=\\"\\" src=\\"http://some.url/someimage.jpg\\"/></figure><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with the "conversation" style 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"conversation\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__text-bubble\\"><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-back\\"></span><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-front\\"></span></div><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div></div></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with the "horizontal" style 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"horizontal\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div></div>
<!-- /wp:coblocks/testimonial -->"
`;
